### DISCUSSION TITLE
DOC: How to add a Jetbrains launcher icon to Workspace

### DISCUSSION THREAD

**Discussion by djarbz** [coder/coder#7729] [General] [answered]
I have icons for Coder, VSCode, SSH, etc...
But what I really want is an icon for Jetbrains Gateway.

It looks like we can define the `coder_app` [resource](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/app).
```terraform
resource "coder_app" "intellij" {
  agent_id     = coder_agent.dev.id
  icon         = "${data.coder_workspace.me.access_url}/icon/intellij.svg"
  slug         = "intellij"
  display_name = "JetBrains IntelliJ"
  command      = "projector run"
}
```

What I am less clear on is how to use the Jetbrains-Gateway registered protocol to auto-launch the Gateway client for this workspace.
Per the documentation, a [Gateway Link](https://www.jetbrains.com/help/idea/remote-development-a.html#space_integration:~:text=the%20welcome%20page.-,Gateway%20link,-%3A%20jetbrains%2Dgateway%3A//connect) can be used to auto-launch the local application and should look something like this URL, with the query deconstructed for readability.
```
jetbrains-gateway://${data.coder_workspace.me.access_url}#
idePath={PathToIDEBinaryInWorkspace}
&projectPath={PathToWorkspaceProjectFolder}
&host={CoderServerHost}
&port=22
&user=coder
&type=ssh
&deploy=false
```

I have been unable to find much documentation on the `jetbrains-gateway` protocol and I'm thinking we may need to set the type to reference the coder plugin?

**Comment by sharkymark** [comment]
@kylecarbs @bpmct

**Reply by matifali** [reply]
bump 🚀

**Comment by matifali** [comment]
This is being tracked in https://github.com/coder/jetbrains-coder/issues/149

**Comment by matifali** [comment, marked as answer]
You can use https://registry.coder.com/modules/jetbrains-gateway for this
